File: app/src/main/java/io/euphoria/xkcd/app/connection/ConnectionListener.java
package io.euphoria.xkcd.app.connection;

import io.euphoria.xkcd.app.connection.event.CloseEvent;
import io.euphoria.xkcd.app.connection.event.IdentityEvent;
import io.euphoria.xkcd.app.connection.event.LogEvent;
import io.euphoria.xkcd.app.connection.event.MessageEvent;
import io.euphoria.xkcd.app.connection.event.NickChangeEvent;
import io.euphoria.xkcd.app.connection.event.PresenceChangeEvent;

/** Created by Xyzzy on 2017-02-24. */

/* Receiver interface for connection events */
public interface ConnectionListener {

    /* Discovered our identity */
    void onIdentity(IdentityEvent evt);

    /* Someone (or we) changed their nick */
    void onNickChange(NickChangeEvent evt);

    /* Someone (or we) posted a (new) message */
    void onMessage(MessageEvent evt);

    /* Someone (or we) joined/left */
    void onPresenceChange(PresenceChangeEvent evt);

    /* Backend delivers messages (on demand or not) */
    void onLogEvent(LogEvent evt);

    /* The backend (or we) closed the connection */
    void onClose(CloseEvent evt);

}

File: app/src/main/java/io/euphoria/xkcd/app/connection/ConnectionListener.kt
package io.euphoria.xkcd.app.connection

import io.euphoria.xkcd.app.connection.event.*

/** Created by Xyzzy on 2017-02-24.  */ /* Receiver interface for connection events */
open interface ConnectionListener {
    /* Connection actually established */
    fun onOpen(evt: OpenEvent?)

    /* Discovered our identity */
    fun onIdentity(evt: IdentityEvent)

    /* Someone (or we) changed their nick */
    fun onNickChange(evt: NickChangeEvent)

    /* Someone (or we) posted a (new) message */
    fun onMessage(evt: MessageEvent)

    /* Someone (or we) joined/left */
    fun onPresenceChange(evt: PresenceChangeEvent)

    /* Backend delivers messages (on demand or not) */
    fun onLogEvent(evt: LogEvent)

    /* The backend (or we) closed the connection */
    fun onClose(evt: CloseEvent)
}
